databuilder/query_language.py

```diff
@@ -110,6 +110,9 @@ def is_in(self, other):
         # The query model requires an immutable Set type for containment queries, but
         # that's a bit user-unfriendly so we accept other types here and convert them
         if isinstance(other, (tuple, list, set)):
+            # Parse any ISO date strings in collection
+            t = type(other)
```

Comment:
You don't need to worry about preserving the collection type here: everything gets transformed into `frozenset` at the next step anyway.